src/beneuro_data/conversion/convert_nwb_to_pyaldata.py

```diff
@@ -0,0 +1,815 @@
+    #   channels. I am not currently loading the channel_map.npy file since for Neuropixels 1.0
+    #   it is simply an array of 0 to 383
+
+    # Get max amplitude channel based on templates
+    chan_best = (templates**2).sum(axis=1).argmax(axis=-1)
+
+    # Get brain area channel map for this specific probe
+    electrode_info_df = electrode_info.to_dataframe()
+    probe_electrode_locations_df = electrode_info_df[
+        electrode_info_df["group_name"] == probe_units.name.split("_")[-1]
+    ]
+    probe_channel_map = probe_electrode_locations_df["location"].to_dict()
+
+    no_pinpoint_channel_map = all(value == "nan" for value in probe_channel_map.values())
+
+    brain_area_spikes_and_chan_best = {}
+    if no_pinpoint_channel_map:
+        brain_areas = {"all"}
+    else:
+        brain_areas = {
+            value for value in probe_channel_map.values() if value not in ["out", "void"]
+        }
+
+    for brain_area in brain_areas:
+
+        if no_pinpoint_channel_map:  # Take all channels if there is no channel map
+            brain_area_channels = [key for key, value in probe_channel_map.items()]
+        else:
+            brain_area_channels = [
+                key for key, value in probe_channel_map.items() if value == brain_area
+            ]
+
+        brain_area_neurons = np.where(np.isin(chan_best, brain_area_channels))[0]
+
+        # Define unit guide
+        unsorted_chan_best = chan_best[brain_area_neurons]
+        sorted_chan_best_indices = np.argsort(
+            unsorted_chan_best
+        )  # Variable with sorted indices
+        sorted_chan_best = unsorted_chan_best[sorted_chan_best_indices]
+
+        # Take neurons that are brain area specific and them sort them according to unit guide
+        brain_area_spikes_and_chan_best[brain_area.replace("-", "_")] = {
+            "spikes": binned_spikes[brain_area_neurons, :][sorted_chan_best_indices, :]
+        }
+        brain_area_spikes_and_chan_best[brain_area.replace("-", "_")][
+            "chan_best"
+        ] = sorted_chan_best
+        brain_area_spikes_and_chan_best[brain_area.replace("-", "_")]["KSLabel"] = (
+            probe_units.KSLabel[brain_area_neurons][sorted_chan_best_indices]
+        )
+
+    return brain_area_spikes_and_chan_best
+
+
+def _parse_pose_estimation_series(pose_est_series: PoseEstimationSeries) -> pd.DataFrame:
+    """
+    Parse pose estimation series data from anipose output
+
+    Parameters
+    ----------
+    pose_est_series :
+        ndx_pose object to parse from nwb file
+
+    Returns
+    -------
+    pd.DataFrame :
+        Contains x, y, z or angle and timestamps
+
+    """
+
+    if pose_est_series.data[:].shape[1] == 3:
+        colnames = ["x", "y", "z"]
+    elif pose_est_series.data[:].shape[1] == 2 and all(pose_est_series.data[:, 1] == 0):
+        # If this is true we assume we are dealing with angle data
+        colnames = ["angle"]
```

Comment:
Don't these `PoseEstimationSeries` objects have a name?
Do you never have 2D data with x-y only?

Reply:
I think it will always be either 3D xyz, or an angle. Thats the end goal of the pose estimation pipeline genji made. And the PoseEstimationSeries is an NWB class i think.